doc/links.md

```diff
@@ -136,6 +136,44 @@ and have that compiled into the library. This is necessary if you want to use
 the python bindings, since they rely on dynamically loading the datamodel
 libraries.
 
+## The `LinkNavigator` utility
+
+`podio::LinkCollection`s store each link separately even if a given object is
+present in several links. Additionally, they don't offer any really easy way to
+look up objects that are linked (apart from manually looping and comparing
+elements). To alleviate these issues, we provide the `podio::LinkNavigator`
+utility class that facilitates navigating links and lookups. It can be
+constructed from any `podio::LinkCollection` and can then be used to retrieve
+linked objects. E.g.
+
+```cpp
+const auto& recoMcLinks = event.get<edm4hep::RecoMCParticleLinkCollection>("RecoMCLinks");
+const auto linkNavigator = podio::LinkNavigator(recoMcLinks);
+
+// For podio::LinkCollections with disparate types just use getLinked
+const auto linkedRecs = linkNavigator.getLinked(mcParticle);
+```
+
+If you want to be explicit about the lookup direction, e.g. in case you have a
+link that has the same `From` and `To` type, you can use the overloads that take
+a second *tag argument*:
+```cpp
+const auto linkedMCs = linkNavigator.getLinked(recoParticle, podio::ReturnTo);
+```
+
+The return type of all methods is a `std::vector<WeightedObject>`, where the
```

Comment:
```suggestion
The return type of all methods is a `std::vector<podio::detail::links::WeightedObject>`, where the
```
With the implicit question whether we should lift this object out of the (implicitly private) `detail` namespace, since it is in principle user facing.

Reply:
Actually we pull the `WeightedObject` into the `podio` top level namespace, so it is in fact actually public.